src/app/clusters/temperature-measurement-server/temperature-measurement-server.cpp

```diff
@@ -15,78 +15,95 @@
+}
 
-// TODO: There's no header that declares this event handler, and it's not 100%
-// clear where best to declare it.
-// https://github.com/project-chip/connectedhomeip/issues/3619
-void emberAfPluginTemperatureMeasurementServerReadEventHandler() {}
+EmberAfStatus emberAfTemperatureMeasurementClusterGetMinMeasuredValue(chip::EndpointId endpoint, int16_t * minMeasuredValue)
+{
+    return emberAfReadAttribute(endpoint, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_TEMP_MIN_MEASURED_VALUE_ATTRIBUTE_ID,
+                                CLUSTER_MASK_SERVER, (uint8_t *) minMeasuredValue, sizeof(*minMeasuredValue), NULL);
+}
 
-void emberAfPluginTemperatureMeasurementServerStackStatusCallback(EmberStatus status) {}
+EmberAfStatus emberAfTemperatureMeasurementClusterGetMaxMeasuredValue(chip::EndpointId endpoint, int16_t * maxMeasuredValue)
+{
+    return emberAfReadAttribute(endpoint, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_TEMP_MAX_MEASURED_VALUE_ATTRIBUTE_ID,
+                                CLUSTER_MASK_SERVER, (uint8_t *) maxMeasuredValue, sizeof(*maxMeasuredValue), NULL);
+}
 
-// -------------------------------------------------------------------------
-// ****** callback section *******
+void emberAfTemperatureMeasurementClusterServerInitCallback(chip::EndpointId endpoint)
+{
+    int16_t value        = 0;
+    EmberAfStatus status = emberAfTemperatureMeasurementClusterGetMeasuredValue(endpoint, &value);
 
-void emberAfPluginTemperatureMeasurementServerInitCallback(void)
+    if (EMBER_ZCL_STATUS_SUCCESS != status)
+    {
+        (void) emberAfTemperatureMeasurementClusterSetMeasuredValueCallback(endpoint, 0);
+    }
+
+    status = emberAfTemperatureMeasurementClusterGetMinMeasuredValue(endpoint, &value);
+
+    if (EMBER_ZCL_STATUS_SUCCESS != status)
+    {
+        (void) emberAfTemperatureMeasurementClusterSetMinMeasuredValueCallback(endpoint, INT16_MIN);
+    }
+
+    status = emberAfTemperatureMeasurementClusterGetMaxMeasuredValue(endpoint, &value);
+
+    if (EMBER_ZCL_STATUS_SUCCESS != status)
+    {
+        (void) emberAfTemperatureMeasurementClusterSetMaxMeasuredValueCallback(endpoint, INT16_MIN);
+    }
+}
+
+EmberAfStatus emberAfTemperatureMeasurementClusterSetMeasuredValueCallback(chip::EndpointId endpoint, int16_t measuredValue)
 {
-    EmberAfStatus status;
-    // FIXME Use real values for the temperature sensor polling the sensor using the
-    //       EMBER_AF_PLUGIN_TEMPERATURE_MEASUREMENT_SERVER_MAX_MEASUREMENT_FREQUENCY_S macro
-    EndpointId endpointId = 1; // Hardcoded to 1 for now
-    int16_t newValue      = 0x1234;
-
-    status = emberAfWriteAttribute(endpointId, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_CURRENT_TEMPERATURE_ATTRIBUTE_ID,
-                                   CLUSTER_MASK_SERVER, (uint8_t *) &newValue, ZCL_INT16S_ATTRIBUTE_TYPE);
-    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    EmberAfStatus status = emberAfWriteAttribute(endpoint, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_TEMP_MEASURED_VALUE_ATTRIBUTE_ID,
```

Comment:
Should we check that value is between `MinMeasureValue` and `MaxMeasuredValue` or `0x8000` (unknown) ?

One way to get this feature without any efforts would be to update the `EmberAfAttributeMetadata` of the attribute to points to the value of `MinMeasureValue` and `MaxMeasureValue`, based on the definition from the XML cluster definition, but I don't think this is supported at the moment.

Reply:
> I think we should leverage the `GENERATED_ATTRIBUTES` table which is generated in endpoints_config.h for the default min max values. And also add this checks as part of the general code in the attribute storage?

I do agree. But at the moment the min/max value from the attribute store only support absolute values, not values relative to others attributes.